### PR TITLE
Allow the type name for ProvideApplicationPartFactoryAttribute to be specified.

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.GenerateAssemblyInfo.targets [hunk 1]
```diff
@@ -76,6 +76,9 @@ Copyright (c) .NET Foundation. All rights reserved.
       <RazorAssemblyAttribute Include="System.Reflection.AssemblyProductAttribute" Condition="'$(Product)' != '' and '$(GenerateAssemblyProductAttribute)' == 'true'">
         <_Parameter1>$(Product)</_Parameter1>
       </RazorAssemblyAttribute>
+      <RazorAssemblyAttribute Include="System.Resources.NeutralResourcesLanguageAttribute" Condition="'$(NeutralLanguage)' != '' and '$(GenerateNeutralResourcesLanguageAttribute)' == 'true'">
```
Comment (pranavkm) [inline src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.GenerateAssemblyInfo.targets]: I missed this when I was generating attributes. Noticed this when diffing Identity's output.

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.Extensions.props [hunk 2]
```diff
@@ -11,6 +11,12 @@
     -->
     <RazorDefaultConfiguration Condition="'$(RazorDefaultConfiguration)'==''">MVC-2.1</RazorDefaultConfiguration>
 
+    <!--
+      Determines the load behavior for Razor compiled views in an Mvc application. By default, the compiled assembly is initialized as an Mvc application part.
+      This results in compiled views being discovered and routes for compiled Razor Pages being set up. Setting this value to false, disables this behavior.
+    -->
+    <EnableDefaultCompiledViewAssemblyLoadBehavior Condition="'$(EnableDefaultCompiledViewAssemblyLoadBehavior)' == ''">true</EnableDefaultCompiledViewAssemblyLoadBehavior>
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.Extensions.props]: @javiercn

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.Extensions.targets [hunk 3]
```diff
@@ -4,23 +4,31 @@
     <!--
       MVC will generally want to add support for runtime compilation, but only for applications. 
     -->
-    <GenerateRazorAssemblyInfo Condition="'$(GenerateRazorAssemblyInfo)'=='' and '$(OutputType)'=='Exe'">true</GenerateRazorAssemblyInfo>
+    <GenerateRazorAssemblyInfo Condition="'$(GenerateRazorAssemblyInfo)'==''">true</GenerateRazorAssemblyInfo>
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.Extensions/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.Extensions.targets]: @rynowak we need to be able to generate `RelatedAssemblyPartAttribute` on all types of assemblies. Missed that in the intiial PR. I repurposed the attribute and added an additional attribute to specifically mean hosting attributes. We could have an alternate naming scheme too.

File: src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets [hunk 4]
```diff
@@ -247,7 +247,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Target 
     Name="RazorGetAssemblyAttributes"
     AfterTargets="GetAssemblyAttributes"
-    Condition="'$(GenerateRazorAssemblyInfo)'=='true' and '$(RazorDefaultConfiguration)'!=''"
+    Condition="'$(GenerateRazorHostAssemblyInfo)'=='true' and '$(RazorDefaultConfiguration)'!=''"
```
Comment (ajaybhargavb) [inline src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets]: Shouldn't this be `GenerateRazorHostingAssemblyInfo`?

Comment (pranavkm) [reply src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets]: Yup.

Comment (ajaybhargavb) [reply src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets]: Add a test?

File: src/Microsoft.AspNetCore.Mvc.Razor.Extensions/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.Extensions.targets (hunk 3 above)
Comment (rynowak) [reply src/Microsoft.AspNetCore.Mvc.Razor.Extensions/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.Extensions.targets]: Sounds good.

**Pull request merged**